jetty-ee11/jetty-ee11-servlet/src/test/java/org/eclipse/jetty/ee11/servlet/ServletContextHandlerTest.java

```diff
@@ -2662,4 +2670,40 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
         HttpTester.Response response = HttpTester.parseResponse(rawResponse);
         assertThat(response.getContent(), containsString("OK2"));
     }
+
+    @Test
+    @EnabledOnOs({LINUX, MAC})
+    public void testGetResource() throws Exception
+    {
+        File baseDir = MavenTestingUtils.getTargetTestingDir(ServletContextHandlerTest.class.getSimpleName());
+        FS.ensureEmpty(baseDir);
+        File index = new File(baseDir, "index.html");
+        index.createNewFile();
+        File sub = new File(baseDir, "subdir");
+        sub.mkdir();
+        File data = new File(sub, "data.txt");
+        data.createNewFile();
+
+        ServletContextHandler context = new ServletContextHandler("/c1", ServletContextHandler.NO_SESSIONS);
+        context.setBaseResourceAsPath(baseDir.toPath());
+        _server.setHandler(context);
+
+        assertThat(context.getServletContext().getResource("/"), notNullValue());
+        assertThat(context.getServletContext().getResource("/index.html"), notNullValue());
+        assertThat(context.getServletContext().getResource("/doesnotexist.html"), nullValue());
+        assertThat(context.getServletContext().getResource("/unknowndir/"), nullValue());
+        assertThat(context.getServletContext().getResource("/subdir/data.txt"), notNullValue());
+        assertThat(context.getServletContext().getResource("/subdir%5Cdata.txt"), nullValue()); //encoded slosh
+        assertThat(context.getServletContext().getResource("/subdir/data.txt%00"), nullValue()); //encoded null
+        //NOTE that these tests return a value, whereas in versions prior to 12 they did not
+        assertThat(context.getServletContext().getResource("//subdir/data.txt"), notNullValue());
+        assertThat(context.getServletContext().getResource("/subdir//data.txt"), notNullValue());
+        assertThat(context.getServletContext().getResource("/subdir%2Fdata.txt"), notNullValue()); //encoded slash
```

Comment:
I need to check what that return value is and if it is an alias before I can OK this.
Note that we don't need comments like the above in the code. Put it as a PR comment instead.

Reply:
I think this is good as the javadoc for this method now says in Servlet 6.1:
>This method bypasses both implicit (no direct access to WEB-INF or META-INF) and explicit (defined by the web application) security constraints. Care should be taken both when constructing the path (e. g. avoid unsanitized user provided data) and when using the result not to create a security vulnerability in the application.
>
>The provided path parameter is canonicalized as per Servlet 6.0, 3.5.2 before being used to match resources.

So it is explicitly warning against aliases.

I think we no longer use this pathway for our own resource serving, so we are no longer reliant on the alias checking here.